### PR TITLE
[Validator] fix LegacyTranslatorProxy

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #31161
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Validator/Util/LegacyTranslatorProxy.php [hunk 1]
```diff
@@ -22,15 +22,26 @@ class LegacyTranslatorProxy implements LegacyTranslatorInterface, TranslatorInte
...
+    /**
+     * @param LegacyTranslatorInterface|TranslatorInterface $translator
+     */
+    public function __construct($translator)
     {
+        if (!$translator instanceof LegacyTranslatorInterface && !$translator instanceof TranslatorInterface) {
+            throw new \InvalidArgumentException(sprintf('The translator passed to "%s()" must implement "%s" or "%s".', __METHOD__, TranslatorInterface::class, LegacyTranslatorInterface::class));
+        }
+
         if (!$translator instanceof LocaleAwareInterface) {
```
Comment (stof) [inline src/Symfony/Component/Validator/Util/LegacyTranslatorProxy.php]: it should implement either `LocaleAwareInterface` or `LegacyTranslatorInterface` though (as LegacyTranslatorInterface also has `setLocale`)

Comment (nicolas-grekas) [reply src/Symfony/Component/Validator/Util/LegacyTranslatorProxy.php]: updated

**Review by stof — APPROVED** [review]

**Comment by pamil** [comment]
Thanks, Nicolas! :)

**Comment by StephanePate** [comment]
@nicolas-grekas when will it be merged into v4.2.7 ? thanks !

**Comment by dvdknaap** [comment]
@nicolas-grekas Thanks for this fix!

**Pull request merged**